sqldelight-compiler/src/main/kotlin/app/cash/sqldelight/core/compiler/QueryGenerator.kt

```diff
@@ -76,6 +78,11 @@ abstract class QueryGenerator(
     statement: PsiElement,
     id: Int
   ): CodeBlock {
+    if (generateAsync) {
+      check(dialect.asyncPreparedStatementType != null) { "Dialect $dialect does not support async drivers" }
+    }
```

Comment:
If you promote the default types up then you dont need these checks, which I think is fine if by default dialects request the interface types.

Reply:
I guess H2 is a problem because it would want to use APIs that dont exist in the interface type. FUN! I'd still promote it just to clean up the code, and we can worry about H2 when it pops up later.